### PR TITLE
feat: add --include-global support to proto use

### LINKED ISSUE #533: Add `--include-global` support to `proto use`
Right now `proto use` only installs tools from the current working directory and upwards, but does not include the global `~/.proto/prototools`. We should add an argument to also include global versions.

Work:
- Update https://github.com/moonrepo/proto/blob/master/crates/cli/src/commands/install_all.rs
- Add arguments
- Rework line 24 (refer to outdated for an example)

### PR DESCRIPTION
**Pull request by varshith257**
Fixes #533 
/claim #533

### CODE REVIEW

**Comment by varshith257** [comment]
@milesj I have added changes to it. Pleasee review it and any changes required will incorporate them

File: crates/cli/src/main.rs [hunk 1]
```diff
@@ -101,7 +101,7 @@ async fn main() -> MainResult {
             Commands::Uninstall(args) => commands::uninstall(session, args).await,
             Commands::Unpin(args) => commands::unpin(session, args).await,
             Commands::Upgrade => commands::upgrade(session).await,
-            Commands::Use => commands::install_all(session).await,
+            Commands::Use => commands::install_all(session, args).await,
```
Comment (milesj) [inline crates/cli/src/main.rs:104]: You'll need to update the `Use` variant also.

**Comment by milesj** [comment]
Can you add a test also.

File: crates/cli/src/main.rs (hunk 1 above)
Comment (varshith257) [reply crates/cli/src/main.rs:104]: Ok

**Comment by varshith257** [comment]
I will add it

File: crates/cli/src/commands/install_all.rs [hunk 2]
```diff
@@ -96,3 +107,39 @@ pub async fn install_all(session: ProtoSession) -> AppResult {
...
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::app::App;
+    use crate::session::ProtoSession;
+    use starbase::AppResult;
+
+    #[tokio::test]
+    async fn test_install_all_with_global() -> AppResult {
+        let app = App::new();
```
Comment (varshith257) [inline crates/cli/src/commands/install_all.rs:120]: @milesj I am not sure about how we initialize app and test proto.

Ahh, any suggestions you incude here? Are tests is in right direction? I think need to explore proto more( IMO will get experience more on it by contributing more to proto)

Comment (milesj) [reply crates/cli/src/commands/install_all.rs:120]: @varshith257 We use integration tests for these, here's some refs: https://github.com/moonrepo/proto/blob/master/crates/cli/tests/use_test.rs

**Comment by varshith257** [comment]
@milesj Aded integration tests. Have a look at it

**Pull request merged**